### PR TITLE
Gpio overhaul + RBL WiFi Mini

### PR DESCRIPTION
**Pull request by SanderVocke**
Overhauled the GPIO by removing TI's interface layer.

Added support for the RedBearLab WiFi Mini board. Now has separate MTBuild configurations for either this board or the Launchpad.

Updated documentation to include instructions for the RedBearLab WiFi Mini.

### CODE REVIEW

File: leash/mtbuildfile.rb [hunk 1]
```diff
@@ -1,12 +1,23 @@
 application_project :leash, File.dirname(__FILE__) do |app|
 
-  app.add_configuration :Debug,
+  app.add_configuration :Debug_Launchpad,
```
Comment (jerryryle) [inline leash/mtbuildfile.rb]: Consider renaming these configs to simply "Launchpad" and "RBL_WiFiMini". It's unlikely we'll introduce both "Debug" and "Release" configurations of each, so I'd drop that convention for now.

File: leash/mtbuildfile.rb [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
     sources: ['src/**/*.c', 'src/**/*.S'],
     toolchain: toolchain(:arm_none_eabi_gcc,
       linker_script: 'linker.ld',
-      cflags: ['-DUSE_FREERTOS', '-DSIMPLELINK_FREERTOS_MULTITHREAD_FULL'],
+      cflags: ['-DUSE_FREERTOS', '-DSIMPLELINK_FREERTOS_MULTITHREAD_FULL', '-DPINOUT_LAUNCHPAD'],
```
Comment (jerryryle) [inline leash/mtbuildfile.rb]: "PINOUT" is very specific. You might use the more common prefix "BOARD" even if the only difference currently is pinout. As in "BOARD_LAUNCHPAD" and "BOARD_RBL_WIFIMINI".

File: leash/src/common/ui.h [hunk 3]
```diff
@@ -38,4 +37,11 @@ enum user_switches{
 #define LED_ERROR LED_RED
 #define LED_JTAG LED_ORANGE
 
+//pin configurations assertion
```
Comment (jerryryle) [inline leash/src/common/ui.h]: I'd probably put these in a .c file, preferably near where they're used. Or just use #if defined X #elif defined Y #elif #error #endif blocks anywhere that needs to differentiate between the configs.

**Comment by jerryryle** [comment]
Closing with feedback.

**Comment by SanderVocke** [comment]
It now seems like the new board isn't working properly. The symptoms resemble those of a stack or heap overflow - maybe the chip on this WiFi Mini has less RAM. I'll refrain from re-opening this until I can solve it.

**Comment by SanderVocke** [comment]
This chip does not have less RAM, so that's not it. Unfortunately the board does not provide any way to get debug access to the CC3200. I tried modifying it by cutting tracks and re-soldering, but have not had any luck. 

I am afraid this will remain a mystery for the time being.

**Comment by SanderVocke** [comment]
Closing again: the issue about RBL Wifi Mini not working should be resolved before merging.

**Comment by SanderVocke** [comment]
Closing this regardless of the RedBearLab issue. It has been clearly noted in the documentation now, so hopefully people will not run into a nasty surprise.

**Pull request merged**